bugsnag-android-core/src/main/java/com/bugsnag/android/ErrorReader.java

```diff
@@ -238,40 +239,41 @@ private static BugsnagException readException(JsonReader reader) throws IOExcept
     }
 
 
-    private static StackTraceElement[] readStackFrames(JsonReader reader) throws IOException {
-        ArrayList<StackTraceElement> frames = new ArrayList<>();
+    private static List<Map<String, Object>> readStackFrames(JsonReader reader) throws IOException {
+        List<Map<String, Object>> frames = new ArrayList<>();
         reader.beginArray();
         while (reader.hasNext()) {
             frames.add(readStackFrame(reader));
         }
         reader.endArray();
-        return frames.toArray(new StackTraceElement[0]);
+        return frames;
     }
 
-    private static StackTraceElement readStackFrame(JsonReader reader) throws IOException {
-        String method = null;
-        String file = null;
-        int lineNumber = 0;
+    private static Map<String, Object> readStackFrame(JsonReader reader) throws IOException {
+        Map<String, Object> map = new HashMap<>();
         reader.beginObject();
+
         while (reader.hasNext()) {
-            switch (reader.nextName()) {
-                case "method":
-                    method = reader.nextString();
-                    break;
-                case "file":
-                    file = reader.nextString();
+            String key = reader.nextName();
+            Object val = null;
+
+            switch (reader.peek()) {
+                case STRING:
+                    val = reader.nextString();
                     break;
-                case "lineNumber":
-                    lineNumber = reader.nextInt();
+                case NUMBER:
+                    val = reader.nextInt();
```

Comment:
The exception possibly thrown here should be caught - there are no known cases of non-integer values being used here but `JsonToken.NUMBER` is not guaranteed to be an integer. The minimum handling should be to still send the report, though the maximal handling could be to create an internal report for it.